### DISCUSSION TITLE
Accept requests from multiple users

### DISCUSSION THREAD

**Discussion by brilliantAlmaz** [pallets/flask#4838] [Q&A] [answered]
I have a request in my Flask application that performs video processing through a neural network. It will take time for another user to not be able to access the website because Flask is busy doing a lot of calculations.
Is there a way to make my Flask app accept requests from multiple users?

**Comment by ThiefMaster** [comment]
- Do not use the Flask DEV server in case you currently do so
- Offload your long-running jobs to something separate such as a Celery worker

**Comment by ghost** [comment, marked as answer]
For requests that take a long time, you might want to consider starting a background job for them.

The server component that comes with Flask is really only meant for when you are developing your application. Even though it is configured to handle concurrent requests by default.
